keps/sig-api-machinery/3488-cel-admission-control/README.md

```diff
@@ -1065,6 +1065,51 @@ Details to consider:
 
 #### CEL Expression Composition
 
+##### Policy Predicates
+
+The match criteria in bindings are not expected to be able to cover all possible
+ways users may want to scope their policies. For example, there is no way to
+match off of kind, only resource. To provide extensibility for the match critiera
+without requiring modifying every validation rule individually, a global predicate
+system is needed. These predicates contain CEL statements that must be satisfied, otherwise
+the policy will be ignored. In order to keep bindings language-agnostic and to support
+singleton policies, the logic should live in the policy definition resource. To enable
+customization per-binding, the CEL statements should have access to the parameter resource.
+
+Here is an example of a policy definition using policy predicates (under the `customConstraints` field):
+
+```yaml
+apiVersion: admissionregistration.k8s.io/v1alpha1
+kind: ValidatingAdmissionPolicy
+metadata:
+  name: "replicalimit-policy.example.com"
+Spec:
+  failurePolicy: Fail
+  paramKind:
+    apiVersion: rules.example.com/v1
+    kind: ReplicaLimit
+  matchConstraints:
+    resourceRules:
+    - apiGroups:   ["apps"]
+      apiVersions: ["v1"]
+      operations:  ["CREATE", "UPDATE"]
+      resources:   ["deployments"]
+  customConstraints:
+    - expression: 'metadata.kind == "Deployment"'
+    - expression: '!(metadata.namespace in params.excludedNamespaces)'
```

Comment:
Note that these expressions do not have an associated `message` field?

Reply:
+1 I can add the fact that no `message` field exists for this. IMO this makes sense, as messages::violations should be 1::1, and this is more a predicate for whether violations will be thrown at all.